third-party/realdds/include/realdds/dds-topic-reader.h

```diff
@@ -72,7 +73,7 @@ class dds_topic_reader : public eprosima::fastdds::dds::DataReaderListener
     };
 
     // The callbacks should be set before we actually create the underlying DDS objects, so the reader does not
-    void run( qos const & = qos() );
+    virtual void run( qos const & = qos() );
```

Comment:
I think it is better to remove the default parameter. Having default for virtual functions can be error prone because the base default will always be used even if a son changes the default.

We don't use the default in our code now, we pass a parameter everywhere used, and future users can just create the qos object when calling run.

Reply:
>the base default will always be used

Only if use point to the base.

>We don't use the default in our code now

We do... in the watcher, felxible-reader. I'll change them, too...